src/main/java/bio/terra/workspace/common/utils/FlightApplicationContext.java

```diff
@@ -0,0 +1,103 @@
+package bio.terra.workspace.common.utils;
+
+import bio.terra.cloudres.google.billing.CloudBillingClientCow;
+import bio.terra.cloudres.google.cloudresourcemanager.CloudResourceManagerCow;
+import bio.terra.cloudres.google.serviceusage.ServiceUsageCow;
+import bio.terra.workspace.app.configuration.external.GoogleWorkspaceConfiguration;
+import bio.terra.workspace.db.DataReferenceDao;
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.service.buffer.BufferService;
+import bio.terra.workspace.service.iam.SamService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.support.TransactionTemplate;
+
+/**
+ * The purpose of FlightApplicationContext is to provide a clean interface for flights to get access
+ * to singleton Spring components. This avoids the use of dynamic bean lookups in flights and
+ * casting the lookup result. Instead, flights make calls to accessors in this class. Spring will
+ * wire up the underlying methods once at startup avoiding the bean lookup. The objects will be
+ * properly types without casting.
+ */
+@Component
+public class FlightApplicationContext {
+  private final BufferService bufferService;
+  private final CloudBillingClientCow billingClient;
+  private final CloudResourceManagerCow resourceManager;
+  private final DataReferenceDao dataReferenceDao;
+  private final GoogleWorkspaceConfiguration googleWorkspaceConfiguration;
+  private final ObjectMapper objectMapper;
+  private final SamService samService;
+  private final ServiceUsageCow serviceUsage;
+  private final TransactionTemplate transactionTemplate;
+  private final WorkspaceDao workspaceDao;
+
+  @Autowired
+  public FlightApplicationContext(
```

Comment:
I'm concerned this class could get arbitrarily large, and it's building beans that may never be needed at runtime depending on the actual dependency graph.

Reply:
The only beans I put in there are ones referenced in flights, so they are needed at runtime.

Reply:
As of right now  yes, but you'll have to remember to remove the ones that aren't needed anymore.